Components/WebService/Rest/RestConfigurableProducer.php

```diff
@@ -199,14 +201,19 @@ protected function request(ClientInterface $client, array $stepActionParams, arr
                     $responseBody = $this->getSerializer()->deserialize(
                         $responseContent,
                         $params[self::REQUEST_EXPECTED_RESPONSE_TYPE],
-                        $encoding
+                        $params[self::REQUEST_EXPECTED_RESPONSE_FORMAT] ?? $encoding
                     );
                 } catch (RuntimeException $e) {
                     // Assume that the exception is one of the JSON exceptions that will kill the workers
                     if (RestConfigurableProtocol::ENCODING_JSON === $encoding && (JSON_ERROR_SYNTAX != json_last_error())) {
                         throw new UnexpectedValueException($e->getMessage());
                     }
-                    // if it cannot parse the response fallback to the textual content of the body
+
+                    if (!$endpointOptions[RestConfigurableProtocol::OPTION_REQUEST_FALLBACK_ON_ERROR]) {
```

Comment:
Because we know the only options for this can be `true` or `false`, we could define the condition like this
```
if (false === $endpointOptions[RestConfigurableProtocol::OPTION_RESPONSE_FALLBACK_ON_ERROR])
```

Reply:
Done